### ISSUE TITLE
`prefer-array-flat` incorrect fix

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1316]
```js
// subAppLoads: SubAppLoad | SubAppLoad[]
const loads: SubAppLoad[] = [].concat(subAppLoads);

// auto fixed
const loads: SubAppLoad[] = subAppLoads.flat();
```

`subAppLoads` could be non-array, so `subAppLoads.flat()` is wrong.

**Comment by fisker** [comment]
Is this a real world case? Why would anyone write code like this?

**Comment by JounQin** [comment]
Why not?

**Comment by fisker** [comment]
Ah, sorry. I forgot `subAppLoads` can be array or non-array.

**Comment by fisker** [comment]
Maybe we can leave this case to `prefer-spread`. Or use the same logic from that rule.

**Comment by fisker** [comment]
The fix should be,

If it's Array, use `foo.flat()`.
If not Array, skip.
If unknown, suggest
1. `foo.flat()`
1. `[foo]`
1. `Array.isArray(foo) ? foo.flat() : [foo]`

**Comment by fisker** [comment]
Wait, is `[subAppLoads].flat()` correct?

**Comment by fisker** [comment]
```js
[].concat(1)
//-> [1]
[1].flat()
//-> [1]
[].concat([1])
//-> [1]
[[1]].flat()
//-> [1]
[].concat([[1]])
//-> [Array(1)]
[[[1]]].flat()
//-> [Array(1)]
```
